app/controllers/deals_controller.rb

```diff
@@ -0,0 +1,30 @@
+class DealsController < ApplicationController
+  load_and_authorize_resource
+
+  def index
+    redirect_to categories_url
+  end
+
+  def new
+    @deal = Deal.new
+  end
+
+  def create
+    @deal = Deal.new(deal_params)
+    @deal.author = current_user
+
+    respond_to do |format|
+      if @deal.save
+        format.html { redirect_to @deal.categories.first, notice: 'Transaction was successfully added.' }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
```

Comment:
- [ ] Good job adding a notice message when a deal is created successfully, kindly add a notice message for when a deal is not created successfully. This will show an error message telling the user that the deal was not created successfully and improve the user experience.

Reply:
Thank you, Done!

Reply:
This issue is well resolved! ✔️